# 找出今天成交量是昨天的几倍的股票

- 剔除停牌的
- 剔除一字板的
- 剔除新股
- 剔除st股
- TODO

In [1]:
import tushare as ts
dfsh = ts.get_hist_data('sh',start='2016-01-05')
date = list(dfsh.index)
[today,last] = date[0:2]
df = ts.get_today_all() 
lastData = df.loc[:,['code','name','volume']]
lastData.to_json(today+'-unicode.json',orient='records')
lastData.to_json(today+'-gbk.json',orient='records',force_ascii=False) 
df.to_json(today+'-gbk-all.json',orient='records',force_ascii=False) 

[Getting data:]#####################################

In [2]:
import tushare as ts
import pandas as pd
import numpy as np
dfsh = ts.get_hist_data('sh',start='2016-01-05')
date = list(dfsh.index)
[today,last] = date[0:2]
dft = pd.read_json(today+'-gbk-all.json',orient='records', typ='frame',numpy=False) 
dfy = pd.read_json(last+'-gbk-all.json',orient='records', typ='frame',numpy=False) 
# pandas.read_json(path_or_buf=None, orient=None, typ='frame', dtype=True, convert_axes=True, 
# convert_dates=True, keep_default_dates=True, numpy=False, precise_float=False, date_unit=None)¶
# Convert a JSON string to pandas object
tdata = dft.loc[:,['code','name','volume']]
ydata = dfy.loc[:,['code','name','volume']]

data =   pd.concat({'code':tdata.loc[:,['code']], 'name':tdata.loc[:,['name']], 'volume':tdata.loc[:,['volume']],'volume-lastday':ydata.loc[:,['volume']] }, axis=1)
def f(x):
    return 99;
data['rate'] = data.applymap(f).all(axis=1).astype(float)
data.as_matrix()
tt = np.array(data)
data.head()
tt



array([[601900L, u'N\u5357\u65b9', 135712L, 127012L, 1.0],
       [600086L, u'\u4e1c\u65b9\u91d1\u94b0', 44356771L, 42035770L, 1.0],
       [600179L, u'\u9ed1\u5316\u80a1\u4efd', 7063710L, 65589898L, 1.0],
       ..., 
       [2174L, u'\u6e38\u65cf\u7f51\u7edc', 1018100L, 979100L, 1.0],
       [2373L, u'\u5343\u65b9\u79d1\u6280', 628700L, 537800L, 1.0],
       [2040L, u'\u5357 \u4eac \u6e2f', 410400L, 350100L, 1.0]], dtype=object)

In [59]:
import tushare as ts
dfsh = ts.get_hist_data('sh',start='2016-01-05')
date = list(dfsh.index)
[today,last] = date[0:2]
df = ts.get_today_all() 
lastData = df.loc[:,['code','name','volume']]
lastData.to_json(today+'-unicode.json',orient='records')
lastData.to_json(today+'-gbk.json',orient='records',force_ascii=False)
aa = df.head()
aa.columns 
# Index([u'code', u'name', u'changepercent', u'trade', u'open', u'high', u'low',
#        u'settlement', u'volume', u'turnoverratio', u'amount', u'per', u'pb',
#        u'mktcap', u'nmc'],
#       dtype='object')
aa.loc[:,['code','name','volume']]
# code	name	volume
# 0	601900	N南方	123012
# 1	000851	高鸿股份	47358233
# 2	002575	群兴玩具	6021533
# 3	002700	新疆浩源	8798670
# 4	000019	深深宝Ａ	30239132




[Getting data:]#####################################

code  name    volume
0  601900   N南方    125012
1  000851  高鸿股份  65402089
2  002575  群兴玩具   6113982
3  002700  新疆浩源   9004190
4  000019  深深宝Ａ  30315532

In [55]:
import tushare as ts
df = ts.get_hist_data('sh',start='2016-01-05')
date = list(df.index)
[today,last] = date[0:2]
today
last

u'2016-02-04'

# 实时行情
## 一次性获取当前交易所有股票的行情数据（如果是节假日，即为上一交易日，结果显示速度取决于网速）

import tushare as ts

ts.get_today_all()
返回值说明：

code：代码
name:名称
- changepercent:涨跌幅
- trade:现价
- open:开盘价
- high:最高价
- low:最低价
- settlement:昨日收盘价
- volume:成交量
- turnoverratio:换手率
- 结果显示：
- 
-       code    name     changepercent  trade   open   high    low  settlement \
- 0     002738  中矿资源         10.023  19.32  19.32  19.32  19.32       17.56
- 1     300410  正业科技         10.022  25.03  25.03  25.03  25.03       22.75
- 2     002736  国信证券         10.013  16.37  16.37  16.37  16.37       14.88
- 3     300412  迦南科技         10.010  31.54  31.54  31.54  31.54       28.67
- 4     300411  金盾股份         10.007  29.68  29.68  29.68  29.68       26.98
- 5     603636  南威软件         10.006  38.15  38.15  38.15  38.15       34.68
- 6     002664  信质电机         10.004  30.68  29.00  30.68  28.30       27.89
- 7     300367  东方网力         10.004  86.76  78.00  86.76  77.87       78.87
- 8     601299  中国北车         10.000  11.44  11.44  11.44  11.29       10.40
- 9     601880   大连港         10.000   5.72   5.34   5.72   5.22        5.20
- 
-         volume       turnoverratio
- 0        375100        1.25033
- 1         85800        0.57200
- 2       1058925        0.08824
- 3         69400        0.51791
- 4        252220        1.26110
- 5       1374630        5.49852
- 6       6448748        9.32700
- 7       2025030        6.88669
- 8     433453523        4.28056
- 9     323469835        9.61735

In [38]:
import tushare as ts
df = ts.get_today_all() 
df.head()

[Getting data:]#####################################

code  name  changepercent  trade   open   high    low  settlement  \
0  601900   N南方         44.046   8.83   7.36   8.83   7.36        6.13   
1  000851  高鸿股份         10.033  13.49  11.65  13.49  11.65       12.26   
2  002575  群兴玩具         10.033  13.16  11.90  13.16  11.48       11.96   
3  002700  新疆浩源         10.028  11.74  10.80  11.74  10.51       10.67   
4  000019  深深宝Ａ         10.027  20.08  18.02  20.08  17.85       18.25   

     volume  turnoverratio     amount      per     pb         mktcap  \
0    122012        0.07215    1069134   18.396  2.516  723265.300000   
1  46811133        9.59616  592864588  129.091  3.069  797750.386740   
2   5991333        1.03313   75813078  219.333  8.618  774755.520000   
3   8675603        2.77093   98432247   26.089  6.072  495929.157120   
4  30226432       11.98262  575298104  453.273  6.280  604569.009472   

             nmc  
0  149315.300000  
1  658057.451794  
2  763174.720000  
3  367571.628120  
4  506522.754944

In [26]:
import tushare as ts

df = ts.get_stock_basics()
date = df.ix['600848']['timeToMarket'] #上市日期YYYYMMDD
df.head()
 
 

name industry area      pe  outstanding      totals   totalAssets  \
code                                                                         
000001   平安银行       银行   深圳    6.02   1180405.50  1430867.63  2.599060e+08   
000002  万  科Ａ     全国地产   深圳   29.51    970832.75  1103913.25  5.707935e+07   
000004   国农科技     生物制药   深圳  776.48      8387.54     8397.66  3.586395e+04   
000005   世纪星源     区域地产   深圳    0.00     91374.29   105853.68  1.336589e+05   
000006   深振业Ａ     区域地产   深圳   27.29    134148.73   134999.50  1.238693e+06   

        liquidAssets  fixedAssets    reserved  reservedPerShare    esp   bvps  \
code                                                                            
000001          0.00    440600.00  5932600.00              4.15  1.270  10.98   
000002   52000132.00    404016.41   834236.94              0.76  0.621   8.05   
000004      28597.28      3792.01       66.45              0.01  0.028   0.99   
000005      22092.81      2413.78    26976.98              0.25 -0.028   0.70   
000006    1015396.63       362.29    48432.15              0.36  0.205   3.15   

           pb  timeToMarket  
code                         
000001   0.91      19910403  
000002   3.04      19910129  
000004  29.74      19910114  
000005  10.19      19901210  
000006   2.37      19920427

In [27]:
import tushare as ts

df = ts.get_stock_basics()
date = df.ix['600848']['timeToMarket'] #上市日期YYYYMMDD
aa = df.head()
aa.index

Index([u'000001', u'000002', u'000004', u'000005', u'000006'], dtype='object', name=u'code')

In [37]:
import tushare as ts

df = ts.get_stock_basics()
date = df.ix['600848']['timeToMarket'] #上市日期YYYYMMDD
aa = df.head()
aa.columns
aa.index
ticks = list(df.index)
len(ticks)

2813

In [17]:
import pandas as pd
import numpy as np
df = pd.DataFrame(np.random.randn(3, 4))
df.head()
 

0         1         2         3
0 -0.870035  0.651422 -0.424765  0.780625
1 -0.902114 -1.113579  2.787032  0.903848
2 -1.066985  1.368586 -1.139575  0.275840

In [22]:
import pandas as pd
import numpy as np
df = pd.DataFrame(np.random.randn(3, 4)) 
df.index  

Int64Index([0, 1, 2], dtype='int64')

In [23]:
import pandas as pd
import numpy as np
df = pd.DataFrame(np.random.randn(3, 4)) 
df.columns 

Int64Index([0, 1, 2, 3], dtype='int64')

In [20]:
import pandas as pd
import numpy as np
df = pd.DataFrame(np.random.randn(3, 4))
df.shape 

(3, 4)

In [21]:
import pandas as pd
import numpy as np
df = pd.DataFrame(np.random.randn(3, 4))
 
df.describe()

0         1         2         3
count  3.000000  3.000000  3.000000  3.000000
mean  -0.737622 -0.766681  0.296817  0.135529
std    0.791982  0.981093  1.138702  0.265372
min   -1.211355 -1.816584 -0.457070 -0.071154
25%   -1.194773 -1.213426 -0.358126 -0.014100
50%   -1.178192 -0.610267 -0.259183  0.042953
75%   -0.500756 -0.241729  0.673760  0.238871
max    0.176679  0.126809  1.606703  0.434788

In [11]:
import pandas as pd
import numpy as np
df = pd.DataFrame(np.random.randn(3, 4))
df.head()
df.index 
df.columns
df.shape
df.describe()

0         1         2         3
count  3.000000  3.000000  3.000000  3.000000
mean  -0.415225 -0.100973 -0.548033  0.237713
std    0.664289  0.892766  1.400837  1.060933
min   -0.825037 -0.774704 -1.536874 -0.647974
25%   -0.798445 -0.607265 -1.349542 -0.350195
50%   -0.771853 -0.439826 -1.162209 -0.052415
75%   -0.210319  0.235893 -0.053612  0.680556
max    0.351215  0.911612  1.054986  1.413528

# todo

# todo

# todo

# 获取沪深上市公司基本情况
## ts.get_stock_basics()
- code,代码
- name,名称
- industry,所属行业
- area,地区
- pe,市盈率
- outstanding,流通股本
- totals,总股本(万)
- totalAssets,总资产(万)
- liquidAssets,流动资产
- fixedAssets,固定资产
- reserved,公积金
- reservedPerShare,每股公积金
- eps,每股收益
- bvps,每股净资
- pb,市净率
- timeToMarket,上市日期

In [2]:
import tushare as ts

df = ts.get_stock_basics() 
df.head()

name industry area      pe  outstanding      totals   totalAssets  \
code                                                                         
000001   平安银行       银行   深圳    6.02   1180405.50  1430867.63  2.599060e+08   
000002  万  科Ａ     全国地产   深圳   29.51    970832.75  1103913.25  5.707935e+07   
000004   国农科技     生物制药   深圳  776.48      8387.54     8397.66  3.586395e+04   
000005   世纪星源     区域地产   深圳    0.00     91374.29   105853.68  1.336589e+05   
000006   深振业Ａ     区域地产   深圳   27.29    134148.73   134999.50  1.238693e+06   

        liquidAssets  fixedAssets    reserved  reservedPerShare    esp   bvps  \
code                                                                            
000001          0.00    440600.00  5932600.00              4.15  1.270  10.98   
000002   52000132.00    404016.41   834236.94              0.76  0.621   8.05   
000004      28597.28      3792.01       66.45              0.01  0.028   0.99   
000005      22092.81      2413.78    26976.98              0.25 -0.028   0.70   
000006    1015396.63       362.29    48432.15              0.36  0.205   3.15   

           pb  timeToMarket  
code                         
000001   0.91      19910403  
000002   3.04      19910129  
000004  29.74      19910114  
000005  10.19      19901210  
000006   2.37      19920427

# 按年度、季度获取业绩报表数据。数据获取需要一定的时间，网速取决于您的网速，请耐心等待。结果返回的数据属性说明如下：

* code,代码
* name,名称
* eps,每股收益
* eps_yoy,每股收益同比(%)
* bvps,每股净资产
* roe,净资产收益率(%)
* epcf,每股现金流量(元)
* net_profits,净利润(万元)
* profits_yoy,净利润同比(%)
* distrib,分配方案
* report_date,发布日期

调用方法：

## 获取2014年第3季度的业绩报表数据
ts.get_report_data(2014,3)

In [3]:
import tushare as ts

df = ts.get_report_data(2015,3)
df.head()

[Getting data:]#http://vip.stock.finance.sina.com.cn/q/go.php/vFinanceAnalyze/kind/mainindex/index.phtml?s_i=&s_a=&s_c=&reportdate=2015&quarter=3&p=1&num=60
#http://vip.stock.finance.sina.com.cn/q/go.php/vFinanceAnalyze/kind/mainindex/index.phtml?s_i=&s_a=&s_c=&reportdate=2015&quarter=3&p=2&num=60
#http://vip.stock.finance.sina.com.cn/q/go.php/vFinanceAnalyze/kind/mainindex/index.phtml?s_i=&s_a=&s_c=&reportdate=2015&quarter=3&p=3&num=60
#http://vip.stock.finance.sina.com.cn/q/go.php/vFinanceAnalyze/kind/mainindex/index.phtml?s_i=&s_a=&s_c=&reportdate=2015&quarter=3&p=4&num=60
#http://vip.stock.finance.sina.com.cn/q/go.php/vFinanceAnalyze/kind/mainindex/index.phtml?s_i=&s_a=&s_c=&reportdate=2015&quarter=3&p=5&num=60
#http://vip.stock.finance.sina.com.cn/q/go.php/vFinanceAnalyze/kind/mainindex/index.phtml?s_i=&s_a=&s_c=&reportdate=2015&quarter=3&p=6&num=60
#http://vip.stock.finance.sina.com.cn/q/go.php/vFinanceAnalyze/kind/mainindex/index.phtml?s_i=&s_a=&s_c=&reportdate=2015&quarter=3&p=

code  name   eps  eps_yoy  bvps    roe  epcf  net_profits  profits_yoy  \
0  300501  海顺新材  1.10      NaN  6.37  18.45  0.75      4416.10          NaN   
1  300499  高澜股份  0.38     5.56   NaN   6.27   NaN      1887.24         3.64   
2  601900  南方传媒  0.53     3.92   NaN  13.97   NaN     34199.62        18.84   
3  001979  招商蛇口   NaN      NaN   NaN    NaN   NaN    191046.55        11.13   
4  002778  高科石化  0.45     4.65   NaN   7.85   NaN      2999.58         5.56   

  distrib report_date  
0     NaN       01-22  
1     NaN       01-20  
2     NaN       01-20  
3     NaN       12-28  
4     NaN       12-16


# 盈利能力¶
## 按年度、季度获取盈利能力数据，结果返回的数据属性说明如下：

* code,代码
* name,名称
* roe,净资产收益率(%)
* net_profit_ratio,净利率(%)
* gross_profit_rate,毛利率(%)
* net_profits,净利润(万元)
* eps,每股收益
* business_income,营业收入(百万元)
* bips,每股主营业务收入(元)
* 
调用方法：

## 获取2014年第3季度的盈利能力数据
ts.get_profit_data(2014,3)

In [4]:
import tushare as ts

df = ts.get_profit_data(2015,3)
df.head()




[Getting data:]###############################################

code   name    roe  net_profit_ratio  gross_profit_rate  net_profits  \
0  600155   宝硕股份  69.08            118.76             2.7797     286.2728   
1  002568   百润股份  58.55             31.88            77.7464     701.0257   
2  600870  *ST厦华  55.82             22.00             7.5697       6.6496   
3  002506   协鑫集成  54.69              6.56            11.5725     376.6235   
4  000520   长航凤凰  46.64             18.48            12.7587     108.8259   

      eps  business_income    bips  
0  0.6006         241.0433  0.5057  
1  0.7823        2198.4758  2.4536  
2  0.0127          30.2219  0.0577  
3  0.1492        5732.6844  2.2717  
4  0.1075         588.5767  0.5815

# 营运能力¶
## 按年度、季度获取营运能力数据，结果返回的数据属性说明如下：

- code,代码
- name,名称
- arturnover,应收账款周转率(次)
- arturndays,应收账款周转天数(天)
- inventory_turnover,存货周转率(次)
- inventory_days,存货周转天数(天)
- currentasset_turnover,流动资产周转率(次)
- currentasset_days,流动资产周转天数(天)
### 调用方法：

## 获取2014年第3季度的营运能力数据
ts.get_operation_data(2014,3)
### 结果返回：

-       code     name   arturnover arturndays  inventory_turnover  inventory_days  \
- 0     600605   汇通能源     5.7747    62.3409        102298.4414         0.0035
- 1     002649   博彦科技          3        120         37127.2743         0.0097
- 2     002707   众信旅游    20.0313    17.9719           7704.973         0.0467
- 3     002039   黔源电力     7.8636    45.7806          6360.2514         0.0566
- 4     300071   华谊嘉信      1.809    199.005          3002.9073         0.1199
- 5     002357   富临运业    53.2925     6.7552          2947.0547         0.1222
- 6     002400   省广股份     4.7772     75.358           2592.977         0.1388
- 7     000415   渤海租赁     8.1159    44.3574          1608.9875         0.2237
- 8     600897   厦门空港       5.11    70.4501          1093.2727         0.3293
- 9     002238   天威视讯    15.5591    23.1376           903.7139         0.3984
- 10    600270   外运发展     3.8491    93.5284           755.0421         0.4768


# 成长能力
## 按年度、季度获取成长能力数据，结果返回的数据属性说明如下：

- code,代码
- name,名称
- mbrg,主营业务收入增长率(%)
- nprg,净利润增长率(%)
- nav,净资产增长率
- targ,总资产增长率
- epsg,每股收益增长率
- seg,股东权益增长率
### 调用方法：

## 获取2014年第3季度的成长能力数据
ts.get_growth_data(2014,3)
### 结果返回：

-       code   name       mbrg        nprg         nav       targ        epsg  \
- 0     600566   洪城股份  1479.2517  84495.0729    338.3363   200.8906  16133.3333
- 1     600105   永鼎股份    39.5802  24575.3671      8.6028   -34.6496    437.0968
- 2     600598  *ST大荒   -40.8583   7030.3488     10.3021   -38.2057      1112.5
- 3     600483   福能股份   354.7807   6718.1981    671.6278  1085.3524   2234.0807
- 4     000678   襄阳轴承    67.0414   6565.1816      5.0957    22.7679        3400
- 5     600822   上海物贸   -27.9066   6005.3197      1.5399   -16.5748   -102.3377
- 6     300134   大富科技    52.2005   5397.1468      15.932    26.7249        6200
- 7     600730   中国高科   137.1223   4068.2816     67.4825    15.0042   3697.0803
- 8     000985   大庆华科    35.0846   3106.1411      7.3651     7.2652        2920
- 9     300208   恒顺电气   263.2526   2510.8422     12.2521     3.4033   1421.9048
- 10    603077   和邦股份    34.8709   1548.3749     46.0789    60.1127    674.1935


# 偿债能力
## 按年度、季度获取偿债能力数据，结果返回的数据属性说明如下：

- code,代码
- name,名称
- currentratio,流动比率
- quickratio,速动比率
- cashratio,现金比率
- icratio,利息支付倍数
- sheqratio,股东权益比率
- adratio,股东权益增长率
###调用方法：

## 获取2014年第3季度的偿债能力数据
ts.get_debtpaying_data(2014,3)
###结果返回：

-       code     name     currentratio quickratio    cashratio      icratio  \
- 0     002739   万达院线            0          0            0            0
- 1     000892   星美联合    3529.4671  3529.4671  352946.7104     5392.979
- 2     300223   北京君正     132.0156   123.7622    11339.085     105.3182
- 3     002348   高乐股份      46.9939    40.4353     1960.449    -543.8806
- 4     300333   兆日科技      41.7573    39.2037    3817.8843      -240.84
- 5     300042   朗科科技      28.8297    27.1808    2508.1954    -115.0911
- 6     300085    银之杰      31.2141    30.1414    1819.2099      63.2537
- 7     002279   久其软件      20.0396    19.9498     876.2346    -317.0054
- 8     002148   北纬通信      23.0136    22.9742    1083.1275    -294.4272
- 9     002322   理工监测      28.9634    27.0216    1693.6619    -112.0428
- 10    300288   朗玛信息      34.7063    34.7063    2475.7024    -269.7696


# 现金流量
## 按年度、季度获取现金流量数据，结果返回的数据属性说明如下：

- code,代码
- name,名称
- cf_sales,经营现金净流量对销售收入比率
- rateofreturn,资产的经营现金流量回报率
- cf_nm,经营现金净流量与净利润的比率
- cf_liabilities,经营现金净流量对负债比率
- cashflowratio,现金流量比率

## 调用方法：

## 获取2014年第3季度的现金流量数据
ts.get_cashflow_data(2014,3)

###结果返回：

-       code    name    cf_sales   rateofreturn  cf_nm     cf_liabilities  \
- 0     600291  西水股份   236.89       0.4004   25.4636         0.5169
- 1     600485  信威集团  11.7365       0.1141  474.5382         0.5092
- 2     600683  京投银泰  10.4481       0.0816         0         0.0879
- 3     000712  锦龙股份   9.8521       0.0338    1.1299         0.0495
- 4     600715  松辽汽车   8.7834       0.0647         0         0.0635
- 5     300059  东方财富   5.7247       0.4412   30.3246         0.7406
- 6     000668  荣丰控股     4.52       0.0221         0         0.0392
- 7     600773  西藏城投   2.6402       0.1763   56.2188         0.2015
- 8     000068  华控赛格   2.2098       0.1357         0         0.1874
- 9     600215  长春经开   1.9857       0.1104         0         0.2359
- 10    300156  神雾环保   1.6736       0.0509   13.1981         0.1571

# .

Pandas是Python下一个开源数据分析的库，它提供的数据结构DataFrame极大的简化了数据分析过程中一些繁琐操作。 
# 1. 基本使用：
  创建DataFrame. DataFrame是一张二维的表，大家可以把它想象成一张Excel表单或者Sql表。Excel 2007及其以后的版本的最大行数是1048576，最大列数是16384，超过这个规模的数据Excel就会弹出个框框“此文本包含多行文本，无法放置在一个工作表中”。Pandas处理上千万的数据是易如反掌的sh事情，同时随后我们也将看到它比SQL有更强的表达能力，可以做很多复杂的操作，要写的code也更少。 说了一大堆它的好处，要实际感触还得动手码代码。首要的任务就是创建一个DataFrame，它有几种创建方式：
- 列表，序列(pandas.Series), numpy.ndarray的字典
- 二维numpy.ndarray
- 别的DataFrame
- 结构化的记录(structured arrays)
- 其中，我最喜欢的是通过二维ndarray创建DataFrame，因为代码敲得最少：
 <br>
- import pandas as pd
- import numpy as np
- df = pd.DataFrame(np.random.randn(3, 4))
- df
- 0 1 2 3
- 0 0.236175 -0.394792 -0.171866 0.304012
- 1 0.651926 0.989046 0.160389 0.482936
- 2 -1.039824 0.401105 -0.492714 -1.220438
- 当然你还可以参考我的这篇文章从mysql数据库或者csv文件中载入数据到dataframe。
- dataframe中index用来标识行，column标识列，shape表示维度。
-  
- df.index 
- df.columns
- df.shape
- 通过describe方法，我们可以对df中的数据有个大概的了解：
-  
- df.describe()
- 0 1 2 3
- count 3.000000 3.000000 3.000000 3.000000
- mean -0.050574 0.331786 -0.168064 -0.144496
- std 0.881574 0.694518 0.326568 0.936077
- min -1.039824 -0.394792 -0.492714 -1.220438
- 25% -0.401824 0.003156 -0.332290 -0.458213
- 50% 0.236175 0.401105 -0.171866 0.304012
- 75% 0.444051 0.695076 -0.005739 0.393474
- max 0.651926 0.989046 0.160389 0.482936
  2. 数据select, del, update。
  按照列名select:
-  
- df[0]
-  
- 0 0.236175
- 1 0.651926
- 2 -1.039824
  按照行数select:
- 1
- df[:3] #选取前3行
- 按照索引select:
-  
- df.loc[0]
-  
- 0 0.236175
- 1 -0.394792
- 2 -0.171866
- 3 0.304012
  按照行数和列数select:
-  
- df.iloc[3] #选取第3行
- df.iloc[2:4] #选取第2到第3行
- df.iloc[0,1] #选取第0行1列的元素
- dat.iloc[:2, :3] #选取第0行到第1行，第0列到第2列区域内的元素
- df1.iloc[[1,3,5],[1,3]] #选取第1，3，5行，第1，3列区域内的元素
  删除某列：
-  
- del df[0]
- df
- 1 2 3
- 0 -0.394792 -0.171866 0.304012
- 1 0.989046 0.160389 0.482936
- 2 0.401105 -0.492714 -1.220438
  删除某行：
-  
- df.drop(0)
-  
- 1 2 3
- 1 0.989046 0.160389 0.482936
- 2 0.401105 -0.492714 -1.220438
  3.运算。
  基本运算：
-  
- df[4] = df[1] + df[2]
-  
- 1 2 3 4
- 0 -0.394792 -0.171866 0.304012 -0.566659
- 1 0.989046 0.160389 0.482936 1.149435
- 2 0.401105 -0.492714 -1.220438 -0.091609
- map运算，和python中的map有些类似：
-  
- df[4].map(int)
- 0 0
- 1 1
- 2 0
- apply运算：
-  
- df.apply(sum)
-  
- 1 0.995359
- 2 -0.504192
- 3 -0.433489
- 4 0.491167
  4. Group by 操作。
  pandas中的group by 操作是我的最爱，不用把数据导入excel或者mysql就可以进行灵活的group by 操作，简化了分析过程。
-  
- df[0] = ['A', 'A', 'B']
- df
-  
- 1 2 3 4 0
- 0 -0.394792 -0.171866 0.304012 -0.566659 A
- 1 0.989046 0.160389 0.482936 1.149435 A
- 2 0.401105 -0.492714 -1.220438 -0.091609 B
-  
- g = df.groupby([0])
-  
- g.size()
-  
- A 2
- B 1
-  
- g.sum()
-  
- 1 2 3 4
- 0
- A 0.594254 -0.011478 0.786948 0.582776
- B 0.401105 -0.492714 -1.220438 -0.091609
- 5. 导出到csv文件
- dataframe可以使用to_csv方法方便地导出到csv文件中，如果数据中含有中文，一般encoding指定为”utf-8″,否则导出时程序会因为不能识别相应的字符串而抛出异常，index指定为False表示不用导出dataframe的index数据。
-  
- df.to_csv(file_path, encoding='utf-8', index=False)